### ISSUE TITLE
Need for `index` entry in CSA header elements?

### ISSUE THREAD

**Issue by matthew-brett** [open-dicom/csa_header#15]
Am I right in thinking that the `index` elements in a CSA header are always sequential?

For example, from the README:

```
{
    'NumberOfPrescans': {'index': 1, 'VR': 'IS', 'VM': 1, 'value': 0},
    'TransmitterCalibration': {'index': 2, 'VR': 'DS', 'VM': 1, 'value': 247.102},
    'PhaseGradientAmplitude': {'index': 3, 'VR': 'DS', 'VM': 1, 'value': 0.0},
    'ReadoutGradientAmplitude': {'index': 4, 'VR': 'DS', 'VM': 1, 'value': 0.0},
    'SelectionGradientAmplitude': {'index': 5, 'VR': 'DS', 'VM': 1, 'value': 0.0},
...
}
```

Are these `index` entries necessary?   I realize they are in the CSA header structure, but, given Python dictionary insertion order, does:

```python
indices = [v['index'] for k, v in parsed_csa.items()]
```

ever differ from `np.arange(len(parsed_csa))`?

**Comment by ZviBaratz** [comment]
Yeah, they definitely don't seem to be necessary. However, it might be a good idea to use `OrderedDict` to make sure order is preserved.

**Comment by matthew-brett** [comment]
Does `OrderedDict` preserve order better than `dict` in Python >= 3.7?

**Comment by ZviBaratz** [comment]
AFAIK, yes. `OrderedDict` still guarantees the order of the dictionary items in ways that `dict` does not.

**Comment by matthew-brett** [comment]
[This page](https://realpython.com/python-ordereddict) lists the advantages of `OrderedDict` as:

* OrderedDict makes it clear that the order of items in the dictionary is important.
 * Features for control over the order of items - you can use `.move_to_end` and `.popitem(last=False)`
* Compare equal iff elements are in same order.

**Comment by ZviBaratz** [comment]
I see. Well, in that case, I guess `dict` would be good enough.
